<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>seev::OriePredSelector</class>
 <widget class="QWidget" name="seev::OriePredSelector">
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <layout class="QVBoxLayout">
   <item>
    <widget class="QComboBox" name="predKind">
     <item>
      <property name="text">
       <string>Always True</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Always False</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Path</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>File Type</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>File Stat</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>File Content</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Accessibility</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>And</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Or</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Inverse</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Match Parent</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Match Children</string>
      </property>
     </item>
     <item>
      <property name="text">
       <string>Prune Dir If (Must use with And)</string>
      </property>
     </item>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="condModifBox">
     <property name="title">
      <string>Condition &amp;&amp; Modifier</string>
     </property>
     <layout class="QHBoxLayout" name="condModifBoxLayout">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="leftMargin">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="rightMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QWidget" name="lhsPredWidg" native="true"/>
      </item>
      <item>
       <widget class="QWidget" name="rhsPredWidg" native="true"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="pathMatchBox">
     <property name="title">
      <string>Path Match</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <property name="verticalSpacing">
       <number>0</number>
      </property>
      <item row="0" column="0">
       <widget class="QRadioButton" name="strstrPathRadio">
        <property name="text">
         <string>String</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QCheckBox" name="icaseBox">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Do case-insensitive searches when checked.&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;Note: &lt;/span&gt;On Windows, all glob patterns are case-&lt;span style=&quot; font-weight:700;&quot;&gt;in&lt;/span&gt;sensitive, regardless of whether ignore case is checked.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string>ICase</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QRadioButton" name="rgxPathRadio">
        <property name="toolTip">
         <string>PCRE2 Regular Expression. Do not use if you do not know what regex is.</string>
        </property>
        <property name="text">
         <string notr="true">RegEx</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QCheckBox" name="fullPathBox">
        <property name="toolTip">
         <string>Search in full path instead of just base name.</string>
        </property>
        <property name="text">
         <string>Full</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QRadioButton" name="fnmatchRadio">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Patterns like &lt;span style=&quot; font-style:italic;&quot;&gt;*.txt &lt;/span&gt;which matches all txt files. &lt;/p&gt;&lt;p&gt;Also support ? (matches a single character) and [] (match a character between).&lt;/p&gt;&lt;p&gt;&lt;span style=&quot; font-style:italic;&quot;&gt;Note: &lt;/span&gt;On Windows, all glob patterns are case-&lt;span style=&quot; font-weight:700;&quot;&gt;in&lt;/span&gt;sensitive, regardless of whether ignore case is checked.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="text">
         <string notr="true">Glob</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QCheckBox" name="lnameBox">
        <property name="toolTip">
         <string>Match against the pointed-to path of a symlink. Implies `-type l`. DO NOT CHECK if you do not know what a symlink is</string>
        </property>
        <property name="text">
         <string>LName</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QRadioButton" name="fuzzPathRadio">
        <property name="text">
         <string>Fuzzy</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QSpinBox" name="fuzzCutPathSpin">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Similarity required for a file to show up in fuzzy matching</string>
        </property>
        <property name="prefix">
         <string>Cutoff: </string>
        </property>
        <property name="singleStep">
         <number>5</number>
        </property>
        <property name="value">
         <number>80</number>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="2">
       <widget class="QLineEdit" name="pathPredEdit">
        <property name="placeholderText">
         <string>The Path to Match Against</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="contMatchBox">
     <property name="title">
      <string>Content Match</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2">
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <property name="verticalSpacing">
       <number>0</number>
      </property>
      <item row="0" column="0">
       <widget class="QRadioButton" name="strstrContRadio">
        <property name="toolTip">
         <string>Simple string match.</string>
        </property>
        <property name="text">
         <string>String</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QCheckBox" name="allowBinContBox">
        <property name="toolTip">
         <string>Also search for non-plain text file.</string>
        </property>
        <property name="text">
         <string>Allow Binary File</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QRadioButton" name="rgxContRadio">
        <property name="toolTip">
         <string>PCRE2 Regular Expression. Do not use if you do not know what regex is.</string>
        </property>
        <property name="text">
         <string notr="true">RegEx</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QCheckBox" name="icaseContBox">
        <property name="toolTip">
         <string>Do case-insensitive match.</string>
        </property>
        <property name="text">
         <string>ICase</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QCheckBox" name="coucurContBox">
        <property name="toolTip">
         <string>Search for content in multiple threads. Will be significantly faster on SSDs and marginally faster on HDDs, but would likely cause CPU usage to soar. Choose according ro your starting paths.</string>
        </property>
        <property name="text">
         <string>Multithreading</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QRadioButton" name="fuzzContRadio">
        <property name="text">
         <string>Fuzzy</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QSpinBox" name="fuzzCutContSpin">
        <property name="enabled">
         <bool>false</bool>
        </property>
        <property name="toolTip">
         <string>Similarity required for a file to show up in fuzzy matching</string>
        </property>
        <property name="prefix">
         <string>Cutoff: </string>
        </property>
        <property name="singleStep">
         <number>5</number>
        </property>
        <property name="value">
         <number>80</number>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="2">
       <widget class="QLineEdit" name="contPredEdit">
        <property name="placeholderText">
         <string>The Content to Search For</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="accessBox">
     <property name="title">
      <string>Accessibility</string>
     </property>
     <layout class="QVBoxLayout" name="_4">
      <property name="spacing">
       <number>0</number>
      </property>
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <item>
       <widget class="QCheckBox" name="readableChk">
        <property name="text">
         <string>Readable</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="writableChk">
        <property name="text">
         <string>Writable</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QCheckBox" name="executableChk">
        <property name="toolTip">
         <string>On Windows &quot;executables&quot; are simply .exe files. (and batch files if you know what they are.)</string>
        </property>
        <property name="text">
         <string>Executable</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="fileTypeBox">
     <property name="title">
      <string>File Type</string>
     </property>
     <layout class="QGridLayout" name="_5">
      <property name="topMargin">
       <number>0</number>
      </property>
      <property name="bottomMargin">
       <number>0</number>
      </property>
      <property name="verticalSpacing">
       <number>0</number>
      </property>
      <item row="3" column="0">
       <widget class="QCheckBox" name="symlinkChk">
        <property name="toolTip">
         <string>Symbolic links. These are NOT .lnk files and if you do not know what a symlink is, chances are that no symlinks exist on your Windows system.</string>
        </property>
        <property name="text">
         <string>Symlink</string>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QCheckBox" name="regFileChk">
        <property name="toolTip">
         <string>Regular, non-directory files.</string>
        </property>
        <property name="text">
         <string>Regular</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QCheckBox" name="dirChk">
        <property name="toolTip">
         <string>Or folders as you like to call them.</string>
        </property>
        <property name="text">
         <string>Directory</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QCheckBox" name="fifoChk">
        <property name="toolTip">
         <string>FIFO files. Do not exist on Windows.</string>
        </property>
        <property name="text">
         <string>Pipe</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QCheckBox" name="blockDevChk">
        <property name="toolTip">
         <string>Block devices. Do not exist on Windows.</string>
        </property>
        <property name="text">
         <string>BlockDev</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QCheckBox" name="charDevChk">
        <property name="toolTip">
         <string>Character Devices. Do not exist on Windows.</string>
        </property>
        <property name="text">
         <string>CharDev</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="fileStatBox">
     <property name="title">
      <string>File Stat</string>
     </property>
     <layout class="QVBoxLayout" name="_6">
      <item>
       <widget class="QComboBox" name="statWhichCbo">
        <item>
         <property name="text">
          <string>Size (MiB)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Size (KiB)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Size (B)</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Accessed Days Before</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Modified Days Before</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Changed Days Before</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Accessed Minutes Before</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Modified Minutes Before</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Changed Minutes Before</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>User ID</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Group ID</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Inode</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="relationBox">
        <property name="title">
         <string>Relation</string>
        </property>
        <layout class="QHBoxLayout" name="_8">
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item>
          <widget class="QRadioButton" name="gtRadio">
           <property name="text">
            <string notr="true">&gt;</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="eqRadio">
           <property name="toolTip">
            <string>For files, all but size are rounded down to the nearest unit before comparing. Size rounds up</string>
           </property>
           <property name="text">
            <string notr="true">==</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QRadioButton" name="ltRadio">
           <property name="text">
            <string notr="true">&lt;</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="QGroupBox" name="targetBox">
        <property name="title">
         <string>Target</string>
        </property>
        <layout class="QFormLayout" name="_7">
         <property name="topMargin">
          <number>0</number>
         </property>
         <property name="bottomMargin">
          <number>0</number>
         </property>
         <item row="0" column="0">
          <widget class="QRadioButton" name="absStatRadio">
           <property name="toolTip">
            <string>When chosen, the respective file stat is compared with the number.</string>
           </property>
           <property name="text">
            <string>Absolute</string>
           </property>
           <property name="checked">
            <bool>true</bool>
           </property>
          </widget>
         </item>
         <item row="0" column="1">
          <widget class="QSpinBox" name="absStatSpn">
           <property name="maximum">
            <number>2147483647</number>
           </property>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QRadioButton" name="relaStatRadio">
           <property name="toolTip">
            <string>When chosen, the respective file stat is compared with that of the target file.</string>
           </property>
           <property name="text">
            <string>Relative</string>
           </property>
          </widget>
         </item>
         <item row="1" column="1">
          <widget class="QLineEdit" name="targEdit">
           <property name="placeholderText">
            <string notr="true">/path/to/target</string>
           </property>
          </widget>
         </item>
         <item row="2" column="0" colspan="2">
          <widget class="QPushButton" name="targBrowseBut">
           <property name="text">
            <string>Browse Target</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>absStatRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>absStatSpn</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>103</x>
     <y>725</y>
    </hint>
    <hint type="destinationlabel">
     <x>208</x>
     <y>725</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>relaStatRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>targBrowseBut</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>76</x>
     <y>736</y>
    </hint>
    <hint type="destinationlabel">
     <x>79</x>
     <y>767</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>relaStatRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>targEdit</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>99</x>
     <y>736</y>
    </hint>
    <hint type="destinationlabel">
     <x>208</x>
     <y>737</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fuzzPathRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>fuzzCutPathSpin</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>74</x>
     <y>181</y>
    </hint>
    <hint type="destinationlabel">
     <x>158</x>
     <y>179</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fuzzPathRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>icaseBox</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>59</x>
     <y>176</y>
    </hint>
    <hint type="destinationlabel">
     <x>112</x>
     <y>114</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fuzzContRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>fuzzCutContSpin</receiver>
   <slot>setEnabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>60</x>
     <y>329</y>
    </hint>
    <hint type="destinationlabel">
     <x>114</x>
     <y>333</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>fuzzContRadio</sender>
   <signal>toggled(bool)</signal>
   <receiver>icaseContBox</receiver>
   <slot>setDisabled(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>58</x>
     <y>334</y>
    </hint>
    <hint type="destinationlabel">
     <x>155</x>
     <y>290</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
